spec/event_sourcery/event_processing/esp_runner_spec.rb

```diff
@@ -25,12 +30,16 @@
     allow(Process).to receive(:wait2).and_return(nil, [pid, success_status])
     allow(Process).to receive(:exit)
     allow(Signal).to receive(:trap)
-    allow(esp_runner).to receive(:shutdown)
+    allow(esp_runner).to receive(:sleep)
   end
 
   describe 'start!' do
     subject(:start!) { esp_runner.start! }
 
+    before do
+      allow(esp_runner).to receive(:shutdown)
```

Comment:
Just skimming this PR at the moment.. so i haven't looked super close at the detail.. but curious as to why you're doing this.. just seems strange to do this on the class under test.

Reply:
Yeah, that is weird. I just moved it from above, from the global context to the 'start!' context. So I could actually test that method!

I'll see why we need that.

Reply:
Ah, so it's used in testing the signal trapping. Just one spec, I'll move the stubbing to that specific context.